site/src/pages/docs/server-multipart.mdx

```diff
@@ -0,0 +1,146 @@
+# Handling a multipart request
+
+Armeria provides <type://Multipart> encoder and decoder built on top of 
+[Reactive Streams](https://www.reactive-streams.org/).
+
+## Building a multipart request
+
+A <type://Multipart> consists of multiple <typeplural://BodyPart>.
+Each <type://BodyPart> is divided into headers and a body. A <type://BodyPart> headers can be created simply 
+using <type://ContentDisposition>. 
+`String`, <type://HttpData>, or <type://StreamMessage> can be set as the body of the <type://BodyPart>.
+
+<Tip>
+
+You can also use <type://HttpHeaders> to set complex headers for <type://BobyPart#headers()>.
+
+</Tip>
+
+```java
+import java.nio.file.Path;
+import com.linecorp.armeria.common.ContentDisposition;
+import com.linecorp.armeria.common.multipart.BodyPart;
+
+// Create a 'Content-Disposition' header with the 'name' parameter set to 'name'.
+ContentDisposition nameDisposition = 
+    ContentDisposition.of("form-data", "name");
+// Create a BodyPart with 'Content-Disposition' header and its data.
+BodyPart bodyPart1 = BodyPart.of(nameDisposition, "Meri Kim");
+
+// Create a 'Content-Disposition' header with the name parameter set to "image", 
+// and the filename parameter set to "profile.png"
+ContentDisposition imageDisposition =
+    ContentDisposition.of("form-data", "image", "profile.png");
+Path image = Paths.get("/path/to/image");
+// Create a BodyPart with 'Content-Disposition' header and its file.
+BodyPart bodyPart2 = BodyPart.of(imageDisposition, StreamMessage.of(image));
+
+// Create a new multipart with the two body parts.
+Multipart multipart = Multipart.of(bodyPart1, bodyPart2);
+```
+
+If we encode and print the <type://Multipart> created above,
+```java
+for (HttpData httpData : multipart.toStreamMessage().collect().join()) {
+  System.out.print(httpData.toStringUtf8());
+}
+```
+we can see how the <type://Multipart> is encoded as shown below:
+```
+--ArmeriaBoundaryEsbNVr9Z66DAIYIN
+content-disposition:form-data; name="name"
+content-type:text/plain
+
+Meri Kim
+--ArmeriaBoundaryEsbNVr9Z66DAIYIN
+content-disposition:form-data; name="image"; filename="profile.png"
+content-type:application/octet-stream
+
+<binary-data>
+--ArmeriaBoundaryEsbNVr9Z66DAIYIN--
+```
+
+## Sending a multipart request
```

Comment:
This is actually a client-side doc. Maybe we should reconsider how we organize our documentation later.

Reply:
Good point. This is client-side documentation. It would be nice to add a category for common features such `HttpHeaders`, `HttpData`, `StreamMessage`, and so on.

Reply:
Yeah, we could probably keep the client-server category for the core features, but other features that require the core knowledge could be put somewhere else.